src/Core/src/Platform/iOS/MauiScrollView.cs

```diff
@@ -14,18 +14,107 @@ public class MauiScrollView : UIScrollView, IUIViewLifeCycleEvents, ICrossPlatfo
 		double _lastArrangeHeight;
 		double _lastArrangeWidth;
 
+		SafeAreaPadding _safeArea = SafeAreaPadding.Empty;
 		UIUserInterfaceLayoutDirection? _previousEffectiveUserInterfaceLayoutDirection;
 
 		WeakReference<ICrossPlatformLayout>? _crossPlatformLayoutReference;
+		WeakReference<IView>? _reference;
 
 		ICrossPlatformLayout? ICrossPlatformLayoutBacking.CrossPlatformLayout
 		{
 			get => _crossPlatformLayoutReference != null && _crossPlatformLayoutReference.TryGetTarget(out var v) ? v : null;
 			set => _crossPlatformLayoutReference = value == null ? null : new WeakReference<ICrossPlatformLayout>(value);
 		}
 
+		internal IView? View
+		{
+			get => _reference != null && _reference.TryGetTarget(out var v) ? v : null;
+			set => _reference = value == null ? null : new(value);
+		}
+
 		internal ICrossPlatformLayout? CrossPlatformLayout => ((ICrossPlatformLayoutBacking)this).CrossPlatformLayout;
 
+		SafeAreaRegions GetSafeAreaRegionForEdge(int edge)
+		{
+			if (View is ISafeAreaPage safeAreaPage)
+			{
+				return safeAreaPage.GetSafeAreaRegionsForEdge(edge);
+			}
+			
+			return SafeAreaRegions.None; // Default: edge-to-edge content
+		}
+
+		SafeAreaEdges? _previousEdges;
+
+		UIEdgeInsets GetInset()
+		{
+			var leftRegion = GetSafeAreaRegionForEdge(0);
+			var topRegion = GetSafeAreaRegionForEdge(1);
+			var rightRegion = GetSafeAreaRegionForEdge(2);
+			var bottomRegion = GetSafeAreaRegionForEdge(3);
+
+			var safeAreaInsets = SafeAreaInsets;
```

Comment:
This code needs to be updated so that if the user has set the region they care about to "SoftInput" or "All" that we add it into the value returned here. 

We need to retrieve the height of the SoftInput and add it to the SafeAreaInset here so that the SCrollView can adjust accordingly